src/userbase-js/types/index.d.ts

```diff
@@ -0,0 +1,70 @@
+/* auth */
+
+export interface Session {
+  sessionId: string
+  creationDate: string
+}
+
+export interface UserProfile {
+  [key: string]: string
+}
+
+export interface UserResult {
+  username: string
+  key: string
+  email?: string
+  profile?: UserProfile
+}
+
+export type KeyNotFoundHandler = (username: string, deviceId: string) => void
+
+export type ShowKeyHandler = (seedString: string, rememberMe: boolean, backUpKey: boolean) => void | Promise<void>
+
+export function init(options: { appId: string, endpoint?: string, keyNotFoundHandler?: KeyNotFoundHandler }): Promise<Session>
```

Comment:
This doesn't seem right now that I'm looking at it (not sure if it changed) but it looks like the return type should be `Promise<{ user: UserResult }>`. Source [this](https://github.com/encrypted-dev/userbase/blob/master/src/userbase-js/auth.js#L509).

Reply:
This is right, but the Session object should actually look like [the result here](https://userbase.dev/docs/sdk/init/). Missed that on my pass through

Reply:
Updated